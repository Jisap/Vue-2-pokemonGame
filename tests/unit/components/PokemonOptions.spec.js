import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOption Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount( PokemonOptions, { 
            props:{
                pokemons:pokemons
            }
        }) 
    })

    test('debe de hacer match con el snapshot', () => {
        expect( wrapper.html()).toMatchSnapshot()
    });
    
    test('debe de mostrar las 4 opciones correctamente', () => {
        const liTags = wrapper.findAll('li')
        expect( liTags.length ).toBe(4)
        expect( liTags[0].text()).toBe('bulbasaur')
        expect( liTags[1].text()).toBe('ivysaur')
        expect( liTags[2].text()).toBe('venusaur')
        expect( liTags[3].text()).toBe('charmander')
    });

    test('debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
        const [li1, li2, li3, li4] = wrapper.findAll('li');     // Identificamos los Li
        li1.trigger('click')                                    // Simulamos un clicks en cada li
        li2.trigger('click') 
        li3.trigger('click') 
        li4.trigger('click') 
        expect( wrapper.emitted('selection').length ).toBe(4)   // Esperariamos que se emitirera 4 eventos 'selection'
        expect( wrapper.emitted('selection')[0]).toEqual([1])   // Esperariamos que el valor enviado por el evento (id) fuese 1
        expect( wrapper.emitted('selection')[1]).toEqual([2])
        expect( wrapper.emitted('selection')[2]).toEqual([3])
        expect( wrapper.emitted('selection')[3]).toEqual([4])
    });                                                         
    
    
})
