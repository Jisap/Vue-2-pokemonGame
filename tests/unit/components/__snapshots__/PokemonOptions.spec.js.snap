// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOption Component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>
      <!--Cuando se haga click en un li se disparar치 el evento selection-->bulbasaur
    </li>
    <li>
      <!--Cuando se haga click en un li se disparar치 el evento selection-->ivysaur
    </li>
    <li>
      <!--Cuando se haga click en un li se disparar치 el evento selection-->venusaur
    </li>
    <li>
      <!--Cuando se haga click en un li se disparar치 el evento selection-->charmander
    </li>
  </ul>
</div>
`;
